docs/proposals/open-ui-command.md

```diff
@@ -0,0 +1,64 @@
+# Add `kubectl` command to open UI
+
+## Motivation
+
+As it was mentioned in [#28007](https://github.com/kubernetes/kubernetes/issues/28007) `kubectl` should have an option to open the UI that is installed under `http://<master_ip>/ui`, for example `kubectl open-ui`. This would be great discoverability improvement for the UI since many people start with CLI, not even knowing that there is a UI.
+
+## Proposed Solution
+
+`kubectl open-ui` command could use different solutions. First one is to run local proxy (`kubectl proxy`) and then open a browser with UI's URL. Second one (not covering all use cases) is to look for UI's service, get it's external endpoint, if it exists, and open a browser with it.
+
+A mix of these two solutions seems to be the best, so if there is an external endpoint of UI's service it can be opened without starting local proxy. Otherwise, proxy can be started to access UI anyways.
+
+###  Command handler's pseudocode
+
+``` go
+func OpenUI(w io.Writer, kubeClient client.Interface) {
+	// Check if UI service exists in the cluster
+	_, err := kubeClient.Services(api.NamespaceSystem).Get(DashboardServiceName)
+	if err != nil {
+		fmt.Printf("Couldn't find UI service in system namespace: %v\n", err)
+		os.Exit(1)
+	}
+
+	// Check if there are any UI service endpoints
+	uiAddress, err := getUIEndpoints(kubeClient)
+	if err != nil {
+		// Run proxy in background
+		...
+	}
+
+    // Run browser with UI's URL
+    ...
+}
+```
+
+### Getting external endpoint of UI's service
+
+First thing, that needs to be verified is availability of UI's service in the cluster. It should be done by checking if there is a `kubernetes-dashboard` service in the `kube-system` namespace. If it cannot be found, then opening UI is not possible and user should be informed about it in returned error.
```

Comment:
I think your question is connected to searching `kubernetes-dashboard` in `kube-system` namespace, which might not be wanted for ordinary user. If it's needed to avoid, that I'd propose to skip this and use solution with proxy, because dashboard will be deployed to `kube-system` in most of the cases:

https://github.com/kubernetes/kubernetes/blob/5debaf7b88a3b6cecb8ce65c5f01568634c09760/cluster/addons/dashboard/dashboard-controller.yaml

What do you think about it?